simpledb/compaction_test.go

```diff
@@ -40,10 +40,49 @@ func TestExecCompactionSameContent(t *testing.T) {
 	assert.Equal(t, "sstable_000000000000042", compactionMeta.ReplacementPath)
 	assert.Equal(t, []string{"sstable_000000000000042", "sstable_000000000000043"}, compactionMeta.SstablePaths)
 
+	// Where is the corresponding Put ???
 	v, err := db.Get("hello")
 	assert.Nil(t, err)
 	assert.Equal(t, "world", v)
 
 	// for cleanups
 	assert.Nil(t, db.sstableManager.currentReader.Close())
 }
+
+func writeSSTableWithTombstoneInDatabaseFolder(t *testing.T, db *DB, p string) {
+	fakeTablePath := filepath.Join(db.basePath, p)
+	assert.Nil(t, os.MkdirAll(fakeTablePath, 0700))
+	mStore := memstore.NewMemStore()
+	assert.Nil(t, mStore.Add([]byte("hello"), []byte("world")))
+	assert.Nil(t, mStore.Delete([]byte("hello")))
```

Comment:
I think you need to write a couple more to have a measurable effect, you can write a long sequence of ascending integers (there should be helpers already) and then delete big chunks from within it. "Punch holes in it" so to say

Reply:
Done in last commit.